### PR TITLE
Make testname selection into a callable function

### PR DESCRIPTION
**Pull request by timholy**
This makes it much easier for CoverageBase to stay in sync with base. See https://github.com/JuliaLang/julia/issues/9493#issuecomment-74193468, which is still happening despite helpful efforts to keep it in sync (https://github.com/timholy/CoverageBase.jl/pull/2).

### CODE REVIEW

**Comment by tkelman** [comment]
Needs rebasing already. Will also cause a lot of churn in every open PR that adds a test. And needs to update CONTRIBUTING.md if the list of tests is going to move.

File: test/choosetests.jl [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        "priorityqueue", "arpack", "file", "version", "resolve",
+        "pollfd", "mpfr", "broadcast", "complex", "socket",
+        "floatapprox", "readdlm", "reflection", "regex", "float16",
+        "combinatorics", "sysinfo", "rounding", "ranges", "mod2pi",
+        "euler", "show", "lineedit", "replcompletions", "repl",
+        "replutil", "sets", "test", "goto", "llvmcall", "grisu",
+        "nullable", "meta", "profile", "libgit2", "docs", "markdown",
+        "base64", "parser", "serialize", "functors", "char", "misc"
+    ]
+
```
Comment (tkelman) [inline test/choosetests.jl]: whitespace here, possibly elsewhere too

**Comment by timholy** [comment]
Of open PRs labeled `test`, only one (#9674) appears to touch `runtests.jl`. I'd be fine if that one gets merged first, but it's not mergable as-is (even without this).

**Comment by andreasnoack** [comment]
#10180 and https://github.com/JuliaLang/julia/commit/4f4dd623993b2bf64d9ecaff8ac3436a3dca314f  touches `runtests.jl` but the the rebases should hopefully be smooth.

**Comment by timholy** [comment]
Those are both in, so I'll rebase mine on top of those.

**Comment by tkelman** [comment]
@timholy I'm curious, did you (or someone else) manually cancel the pending AppVeyor builds of the older commit here, or is this something that AppVeyor is doing for us automatically now? I have code that checks and fails quickly when there are newer pending builds for the same PR, so manually cancelling only saves a minute or so in queue time.

**Comment by timholy** [comment]
I did it manually.

**Comment by tkelman** [comment]
Ah, okay. That isn't generally necessary unless you're really gung-ho about saving the minute or so that my code takes to stop the build.

The latest run here froze on AppVeyor during compilation of C code due to some recent makefile changes (see discussion here https://github.com/JuliaLang/julia/commit/e85a1a5967be8a9910fe333609b75262eb23e794#commitcomment-9756164), so I restarted it.

**Comment by timholy** [comment]
Forgot about that; I won't worry about it in the future.

**Comment by timholy** [comment]
OK, this is green. If I don't hear any objections, I'll merge this very soon (to avoid further churn for other PRs that modify `runtests.jl`).

**Pull request merged**